HW02_B_Mimic_starter.Rmd

```diff
@@ -1,99 +1,125 @@
+#theme_bw() is best theme (IMO)
+
+#for graph 3:
+library("ggrepel")
+
+
+```
+
+## HW02 Part B
+
+For this part of the HW, the goal is to try to recreate the graphs I make from scratch. I will only provide the MD, not the actual code I used to create it besides which data I use to create it. The rest will be up to you. 
+
+Try for all 4, but if you are struggling don't worry about it. Try your best for each, if you don't get everything that's what the peer-review is for. :smile:
+
+
+### Graph 1
+```{r, echo = T, include = TRUE}
+data("diamonds")
+#hint think about the *position* the bars are in...
+```
+
+
+Using the diamonds dataset, make this graph:
+```{r graph1 code, echo=FALSE}
+
+ggplot(diamonds)+
+  geom_bar(aes(cut, ..count.., fill = clarity), position = "dodge") + 
+  labs(title = "My Diamond Collection", x = "Diamond Cut", y = "Number of Diamonds")+
+  theme(plot.title = element_text(hjust = 0.5))
+
+```
+
+### Graph 2
+```{r, echo = T, include = TRUE}
+data("iris")
+  
+
+```
+
+Using the iris dataset, make this graph:
+```{r graph 2 code, echo=FALSE}
+#this is of using the facets
+ggplot(iris, aes(x = Sepal.Length, y = Petal.Length, color = Species, shape = Species)) + 
+  geom_point()+
+  geom_smooth(method = lm)+
+  facet_wrap(factor(iris$Species, levels = c("versicolor","setosa","virginica")))
+```
+
+
+### Graph 3
+You'll need the information in this first box to create the graph
+```{r, echo = TRUE}
+data("mpg")
+corvette <- mpg[mpg$model == "corvette",]
+#install
+require("ggrepel") #useful for making text annotations better, hint hint
+set.seed(42)
+
+```
+
+Now using the mpg dataset and the corvette dataset, make this graph:
+
+```{r graoh 3 code}
+
+#geom_text and geom_text_repel both works for adding notes, but repel helps better viasualition
+ggplot(mpg, aes(x = displ, y = hwy))+
+  geom_point()+
+  geom_point(color = "blue", data = corvette)+
+  geom_text_repel(data = corvette, label = paste("Corvette,",corvette$year))
+
+```
+
+There is a trick to getting the model and year to print off together. `paste()` is a useful function for this, also pasting together parts of file names and parts of urls together. 
+
+
+### Graph 4 
+```{r, echo = T}
+data(mpg)
+
+#hint for the coloring, colorbrewer and you can set palette colors and make your graphs colorblind friendly
+library(RColorBrewer)
+display.brewer.all(colorblindFriendly = T) #take a look at the colorblindfriendly options
+
+```
+
+The above graph lets you see some colobrlind friendly palettes. For the graph below, I used Set2.
+
+
+
+Now using the above mpg dataset, make this graph
+
+```{r graph 4 code}
+
+ggplot(mpg, aes(x = cty, y = class))+
+  geom_point(aes(color = class), position = position_dodge2(width = 0.7))+
+  scale_color_manual(values=brewer.pal(7,"Set2"))+
+  #alpha controls how transpant the plots are
+  geom_boxplot(alpha = 0)+
+  labs(title = "Horizontal BoxPlot of City MPG and Car Class", x = "Car Class", y = "City mpg")
+
```

Comment:
I would refer to the key for this graph; it was tough!

Reply:
Thank you! Is the key the one Robert said would be included in a video?